### DISCUSSION TITLE
How do I disable EMA？

### DISCUSSION THREAD

**Discussion by zlj-ky** [ultralytics/yolov5#7012] [Q&A] [answered]
How do I disable EMA **in my code** so that I can figure out the specific effect of EMA by comparison？

I tried commenting out all the parts about EMA in the train.py, and in this place 

![图片](https://user-images.githubusercontent.com/74093297/158587704-443b1dbe-e588-4d6e-af0b-2aaa2f4c5232.png)

i modify 'model=ema.ema' to 'model=model', and then in my train process, i got an error at the end of first epoch

![图片](https://user-images.githubusercontent.com/74093297/158588046-90df7109-cc36-430f-9b19-a16f1dca059f.png)

how can i fix this bug？

**Comment by glenn-jocher** [comment, marked as answer]
@zlj-ky set decay to zero:
https://github.com/ultralytics/yolov5/blob/c09fb2aa95b6ca86c460aa106e2308805649feb9/utils/torch_utils.py#L286

**Comment by zlj-ky** [comment]
thx for your reply! I want to confirm that just set decay to zero and do not change anything else？
